nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -647,10 +659,13 @@ sub generateXserverConfig {
     mkpath($outDir, 0, 0755);
     write_file($fn, $hwConfig);
 
-    # Generate a basic configuration.nix, unless one already exists.
+    # Generate a basic configuration.nix and flake.nix, unless one already exists.
     $fn = "$outDir/configuration.nix";
-    if ($force || ! -e $fn) {
+    my $flakeFn = "$outDir/flake.nix";
+    if ($force || ! -e $fn || ! -e $flakeFn) {
         print STDERR "writing $fn...\n";
+        print STDERR "writing $flakeFn...\n";
+        write_file($flakeFn, $flakeConfig);
```

Comment:
Isn't flake.nix depending on the folder being a git repo?

Making it a git repo might not be such a bad thing either, it's a good way to encourage users to commit their config.

Reply:
> A flag to `--create-git-repo` would be welcome, but in another PR.

Seems like a task for `;` instead.
Next thing you know you're adding flags to pass flags to git.
It also interferes with the workflow where the user initializes the repo by cloning their remote.
I'd recommend not to add the flag, especially not before `nix flake init` has a flag to do the same, which I think it shouldn't either.